linodecli/baked/operation.py

```diff
@@ -463,10 +463,55 @@ def _add_args_post_put(self, parser) -> List[Tuple[str, str]]:
 
         return list_items
 
+    def _validate_parent_child_conflicts(self, parsed: argparse.Namespace):
+        """
+        This method validates that no child arguments (e.g. --interfaces.purpose) are
+        specified alongside their parent (e.g. --interfaces).
+        """
+        conflicts = {}
+
+        for arg in self.args:
+            parent = arg.parent
+            arg_value = getattr(parsed, arg.path, None)
+
+            if parent is None or arg_value is None:
+                print(arg.path, parent, arg_value)
+                continue
+
+            # Special case to ignore child arguments that are not specified
+            # but are implicitly populated by ListArgumentAction.
+            if isinstance(arg_value, list) and arg_value.count(None) == len(
+                arg_value
+            ):
+                continue
+
+            # If the parent isn't defined, we can
+            # skip this one
+            if getattr(parsed, parent) is None:
+                continue
+
+            if parent not in conflicts:
+                conflicts[parent] = []
+
+            # We found a conflict
+            conflicts[parent].append(arg)
```

Comment:
We might be able to use [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) for `conflicts` to simplified things. Just an optional suggestion.

Reply:
Nice find!